template/web/wp-content/themes/<%= slug %>/app/Repositories/Repository.php

```diff
@@ -1,28 +1,26 @@
 <?php
 /**
- * Parent repository class. Provides a very basic, fluent interface for interacting
- * with PostCollection/PostQuery classes.
+ * Parent repository class. Provides a very basic, fluent interface to interact
+ * with query
  *
  * @package Studiometa
  */
 
 namespace Studiometa\Repositories;
 
-use Timber\PostCollection;
-
 /** Class */
 abstract class Repository {
 	/**
 	 * List of posts.
 	 *
-	 * @var array|PostCollection
+	 * @var array
 	 */
 	private $result_set = array();
 
 	/**
-	 * Returns a list or collection of posts.
+	 * Returns an array of posts.
 	 *
-	 * @return array|PostCollection
```

Comment:
**question:** Should we keep `ArrayObject` as an allowed type?

The `PostQuery` and `PostCollection` classes are base on PHP's `ArrayObject` class. Allowing the usage of `ArrayObject` results could help implement `PostCollection` or `PostQuery` in child classes. 

Maybe we can discuss this next monday?

Reply:
yes, can plan a meeting on the topic ! 
There is no `PostQuery` or `PostCollection` logic in this class since the `do_query()` method is to implement. 
This could lead to some implementation with direct SQL query with `wpdb` for example